src/DataMapper/Product/Provider/ProductIndexedPricesProvider.php

```diff
@@ -0,0 +1,33 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Setono\SyliusMeilisearchPlugin\DataMapper\Product\Provider;
+
+use Sylius\Component\Core\Model\ChannelInterface;
+use Sylius\Component\Core\Model\ProductInterface;
+use Sylius\Component\Core\Model\ProductVariantInterface;
+
+final class ProductIndexedPricesProvider implements ProductIndexedPricesProviderInterface
+{
+    /** We need to provide the price from the first valid variant, to reflect the default Sylius UI behavior */
+    public function getPricesForChannel(ProductInterface $product, ChannelInterface $channel): ProductPrices
```

Comment:
Shouldn't we use the default Sylius service to get the variant also? I.e. the `ProductVariantResolverInterface`